src/compiler/worker/runWebpackCompiler.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2017-present, Callstack.
+ * All rights reserved.
+ * 
+ * @flow
+ */
+
+const EventEmitter = require('events');
+const webpack = require('webpack');
+
+const getConfig = global.requireWithRootDir('./getConfig');
+const Events = global.requireWithRootDir('../events');
+
+module.exports = function runWebpackCompiler({
+  platform,
+  options,
+  fs,
+}: {
+  [key: string]: string,
+  fs: Object,
+}) {
+  const emitter = new EventEmitter();
+
+  const { configPath, configOptions } = JSON.parse(options);
+  const config = getConfig(configPath, configOptions, platform);
+
+  let lastPercent = -1;
+  config.plugins.push(
```

Comment:
Maybe clone the config instead of mutating it? If the user is sharing same config between multiple compilers, this could add multiple plugins to the same config.

Reply:
Doesn't matter - `makeReactNativeConfig` always create a fresh copy of the config.